### PR TITLE
 Hallucinations stop when dead

### LINKED ISSUE #22777: Dead bodies can hallucinate from the supermatter
Tried fixing the supermatter.. got killed and my body continued to hallucinate whilst dead

### PR DESCRIPTION
**Pull request by Supermichael777**
:cl:
fix: Hallucinations stop when dead.
/:cl:

God dammit git.

Fixes #22777.

### CODE REVIEW

File: code/modules/mob/living/carbon/death.dm [hunk 1]
```diff
@@ -7,8 +7,17 @@
 
 	if(!gibbed)
 		emote("deathgasp")
-
+	
+	handling_hal = 1
+	if(halimage)
+		halimage.qdel()
```
Comment (lzimann) [inline code/modules/mob/living/carbon/death.dm]: qdel is a global proc(`qdel(halimage)`)

Comment (Supermichael777) [inline code/modules/mob/living/carbon/death.dm]: fug thanks

File: code/modules/mob/living/carbon/death.dm [hunk 2]
```diff
@@ -7,8 +7,17 @@
...
-
+	
+	handling_hal = 1
+	if(halimage)
+		qdel(halimage)
+	if(halbody)
+		qdel(halbody)
+	if(halitem)
+		qdel(halitem)
+	hal_screwyhud = 0
```
Comment (KorPhaeron) [inline code/modules/mob/living/carbon/death.dm]: Where did you duplicate all this code from/why is there not a proc to clear hallucinations?

**Review by imtakingabreakdontatme — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/carbon/death.dm (hunk 2 above)
Comment (Supermichael777) [inline code/modules/mob/living/carbon/death.dm]: i didn't duplicate it but DNM halucinations are obj/ for some reason

**Comment by Supermichael777** [comment]
The mob has not code way of identifying the hallucinations they own and OTHER PEOPLES HALLUCINATIONS CAN SO I CAN'T EVEN USE REFERENCES TO GET THE LIST! Halucinations need a much bigger refactor then i can give right now

**Pull request closed without merging**